### DISCUSSION TITLE
Compatibility with jesec/rtorrent

### DISCUSSION THREAD

**Discussion by MrStarktastic** [cross-seed/cross-seed#228] [General]
I was able to set up an /RPC2 endpoint that uses Nginx to communicate with rTorrent. The search command works great.
However, daemon mode is not working for me. I followed the instructions from the wiki to make rTorrent curl the cross-seed container. According to rTorrent's execute.log the script is called, but there's nothing happening in cross-seed's container.
Was anyone able to make it work with jesec/rtorrent? It's distroless but it should have curl...
Maybe a different rTorrent image should be used?

**Comment by mmgoodnow** [comment]
To get more information about the source of the problem, I recommend shelling into your rtorrent container and invoking curl manually:
```bash
docker exec -it rtorrent bash
curl -v http://cross-seed:2468
```

If you get an error message from cross-seed, then the problem is likely with rTorrent executing your script. If you still can't connect to cross-seed, then the problem is likely with your docker network configuration. I'll stay tuned

**Reply by MrStarktastic** [reply]
Thank you for the swift response. That's the thing, I can't shell into it because it's distroless (`docker exec -it rtorrent bash/sh/ash` all do not work). I did shell into a temporary ubuntu container that I deployed into the same Docker network and it successfully communicated with cross-seed using the same download-finished.sh script:
```
#!/bin/sh
curl -XPOST http://cross-seed:2468/api/webhook --data-urlencode "name=$1"
exit 0
```
And in rtorrent.rc I have this line in the end:
```
method.set_key=event.download.finished, cross-seed, "execute.throw = {/config/download-finished.sh, $d.name=}"
```
Assuming that the script and config are valid, the jesec/rtorrent image might be the problem here.
I was using that alongside jesec/flood, but perhaps the combined jesec/rtorrent-flood image (which should be shellable) could be the solution. Otherwise, I'll try a different rTorrent Docker image.

**Reply by mmgoodnow** [reply]
is your script chmodded correctly? I've forgotten to do that in the past and rTorrent is not great about error handling so it's hard to tell if it's invoking the script successfully. One other thing you could try is doing something along the lines of `touch  /config/test.tmp` inside the script, and then seeing if that file shows up.

**Reply by MrStarktastic** [reply]
Sorry for the delay... Yes, the script was chmodded correctly and I'm pretty sure that jesec/rtorrent does not even have the "touch" command's binary as I've tried to do something similar to what you suggested. I realized after migrating to rtorrent-flood that curl was not available in there but wget was. So I modified the script to use wget and now it's working (yet to have any results with more than 0 torrents found, but that's a different story).
